### ISSUE TITLE
Possible GL Crash On Context Garbage Collection

### ISSUE THREAD

**Issue by fundamental** [IceDragon200/mruby-nanovg#2]
I've observed on my system nanovg getting into invalid states when what appears to be an old instance of a context gets collected in a garbage collection pass when another context is rendering.
This results in the shader program getting invalidated in glnvg__deleteShader().

I don't know exactly why this is happening as I can't trace this much further myself (from what I can tell that routine should only kill old unused shaders), but leaking the shaders seems to temporarily fix this issue locally. A better workaround might be to add a close method which could clean up an instance deterministically.

**Comment by IceDragon200** [comment]
@fundamental Normally you will only have 1 context for the lifetime of your application.
I never made it a singleton and allowed the initialization of multiple contexts.

tl;dr Use only one context for ALL your rendering

But I'll happily add a `#destroy` method so you'll feel safer :)

**Comment by IceDragon200** [comment]
@fundamental https://github.com/IceDragon200/mruby-nanovg/commit/5f75bf8cc30394fdaa38379825c67311f00335e5 Is this ok? :)

**Comment by fundamental** [comment]
Yeah, that should fix this. In the final app there's only going to be one context, though right now stuff is still in the prototyping phase so I've got a few routines which essentially initialize everything from scratch, run for a short time, and then destroy their resources. Those routines are usually a pretty robust way to test stuff out in a REPL like context, which is why this shader invalidation caught me by surprise.

Thanks for the fix.
